src/nl/hannahsten/texifyidea/util/Labels.kt

```diff
@@ -164,12 +171,17 @@ fun PsiElement.extractLabelName(): String {
     return when (this) {
         is BibtexEntry -> identifier() ?: ""
         is LatexCommands -> {
-            // For now just take the first label name (may be multiple for user defined commands)
-            val info = CommandManager.labelAliasesInfo.getOrDefault(name, null)
-            val position = info?.positions?.firstOrNull() ?: 0
-            val prefix = info?.prefix ?: ""
-            // Skip optional parameters for now (also below and in
-            prefix + this.requiredParameter(position)
+            if (Magic.Command.labelAsParameter.contains(name)) {
```

Comment:
Either do the same for the method `extractLabelElement` below, or figure out a way to remove the code duplication here.
Now if you resolve the reference it ends up at the file which is inputted instead of the label.

Reply:
I think this actually goes together with the "Find Usages" functionality. I will look into it

Reply:
Should we maybe create a separate issue for references to labels in optional parameters? I am not sure whether reference support for these labels will haben any time soon (see Gitter).